share/revng/components/revng.yml

```diff
@@ -0,0 +1,24 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+Pipelines:
+  - share/revng/pipelines/isolate-translate.yml
+  - share/revng/pipelines/analyses-lists.yml
+  - share/revng/pipelines/enforce-abi.yml
+  - share/revng/pipelines/translate.yml
+  - share/revng/pipelines/yield-assembly.yml
+  - share/revng/pipelines/yield-call-graph.yml
+  - share/revng/pipelines/yield-cfg.yml
+Libraries:
+  - lib/revng/analyses/librevngABI.so
+  - lib/revng/analyses/librevngBasicAnalyses.so
+  - lib/revng/analyses/librevngABIAnalyses.so
+  - lib/revng/analyses/librevngEarlyFunctionAnalysis.so
+  - lib/revng/analyses/librevngFunctionIsolation.so
+  - lib/revng/analyses/librevngModel.so
+  - lib/revng/analyses/librevngModelImporter.so
+  - lib/revng/analyses/librevngModelImporterBinary.so
+  - lib/revng/analyses/librevngRecompile.so
+  - lib/revng/analyses/librevngTypeShrinking.so
+  - lib/revng/analyses/librevngYieldPipes.so
```

Comment:
This will go out of sync. We either:
1. Need to generate this in CMake, or...
2. We can we can put in each `share/revng/pipelines/*.yml` file a field to specify the "component", expose it through GraphQL and then let `revng-daemon-self-test` select only the artifacts with the `revng` component. This sounds better, it's more informative for other use cases and does not increase the `dlopen` trickery.

Reply:
1. I'll check if this can be done without too much magic
2. This can work for pipelines, but for libraries it's non-trivial: unless we do `$unusual_things` to them (e.g. stuffing the component in `.note` section)

Overall if (1) can be done in a simple way, it's preferable to (2)

Reply:
We can keep loading the libraries, we just need to prevent it from running revng-c analyses and producing revng-c artifacts.
Simply loading a library won't lead to non-determinism in most cases, I'm fine with that.
Let's strive for option 2.

Reply:
Ok for (2), I've noted down in my pad other options if we want to go back to (1)
Regarding the field: should it be part of the schema or e.g. embedded in a comment?

Reply:
A proper field

Reply:
Done